File: parse_pdf.py
import sys
try:
    from pypdf import PdfReader
except ImportError:
    import subprocess
    subprocess.check_call([sys.executable, "-m", "pip", "install", "pypdf"])
    from pypdf import PdfReader

def extract(path):
    reader = PdfReader(path)
    text = ""
    for page in reader.pages:
        text += page.extract_text() + "\n"
    print(f"--- {path} ---")
    print(text[:1000])
    with open(path.replace('.pdf', '.txt'), 'w', encoding='utf-8') as f:
        f.write(text)
    print(f"Saved to {path.replace('.pdf', '.txt')}")

extract('documents/OFFICIAL BY LAWS_ BREAKPOINT BILLIARDS LEAGUE.pdf')
extract('documents/Breakpoint International Standard (BIS).pdf')

File: parse_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

async function main() {
  const file1 = 'documents/OFFICIAL BY LAWS_ BREAKPOINT BILLIARDS LEAGUE.pdf';
  const file2 = 'documents/Breakpoint International Standard (BIS).pdf';

  try {
    let data1 = await pdf(fs.readFileSync(file1));
    console.log("BYLAWS PAGES:", data1.numpages);
    console.log("BYLAWS PREVIEW:", data1.text.substring(0, 500));

    let data2 = await pdf(fs.readFileSync(file2));
    console.log("BIS PAGES:", data2.numpages);
    console.log("BIS PREVIEW:", data2.text.substring(0, 500));
  } catch (e) {
    console.error("Error reading pdfs:", e);
  }
}

main();
